server/join.go

```diff
@@ -72,9 +73,8 @@ func PrepareJoinCluster(cfg *Config) error {
 	}
 
 	// Cases with data directory.
-
 	initialCluster := ""
-	if wal.Exist(cfg.DataDir) {
+	if wal.Exist(path.Join(cfg.DataDir, "member")) {
```

Comment:
How about check if the directory is empty instead?

Reply:
if empty, wal.Exist() will return false.